### PR TITLE
MMCA 4786 - Upgrade scala & library dependancies to latest version

### PR DESCRIPTION
**Pull request by HariHmrc**
During compilation, below 2 warnings are yet to be sorted.

conf/app.routes:18:1: match may not be exhaustive.
conf/app.routes:19:1: match may not be exhaustive.

### CODE REVIEW

File: test/connectors/CustomsDataStoreConnectorSpec.scala [hunk 1]
```diff
@@ -133,11 +151,13 @@ class CustomsDataStoreConnectorSpec extends SpecBase {
   }
 
   trait Setup {
-    val mockHttpClient = mock[HttpClient]
+    val mockHttpClient = mock[HttpClientV2]
```
Comment (tyagikumarvineet) [inline test/connectors/CustomsDataStoreConnectorSpec.scala:154]: Better to provide the explicit return type as it is a public member

File: test/connectors/CustomsFinancialsApiConnectorSpec.scala [hunk 2]
```diff
@@ -98,10 +117,12 @@ class CustomsFinancialsApiConnectorSpec extends SpecBase {
   }
 
   trait Setup {
-    val mockHttpClient = mock[HttpClient]
+    val mockHttpClient = mock[HttpClientV2]
```
Comment (tyagikumarvineet) [inline test/connectors/CustomsFinancialsApiConnectorSpec.scala:120]: Better to add an explicit return as it is a public member

File: test/connectors/CustomsSessionCacheConnectorSpec.scala [hunk 3]
```diff
@@ -70,9 +81,12 @@ class CustomsSessionCacheConnectorSpec extends SpecBase {
   }
 
   trait Setup {
-    val mockHttpClient = mock[HttpClient]
+    val mockHttpClient = mock[HttpClientV2]
```
Comment (tyagikumarvineet) [inline test/connectors/CustomsSessionCacheConnectorSpec.scala:84]: Better to add an explicit return as it is a public member

File: test/controllers/LanguageSwitchControllerSpec.scala [hunk 4]
```diff
@@ -22,6 +22,7 @@ import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
 
+
```
Comment (tyagikumarvineet) [inline test/controllers/LanguageSwitchControllerSpec.scala:25]: Remove this extra line break

File: test/viewmodels/SecuritiesRequestedStatementsViewModelSpec.scala [hunk 5]
```diff
@@ -262,7 +261,4 @@ trait SpecBaseWithSetup extends SpecBase {
     SecuritiesRequestedStatementsViewModel(securityStatements)
   }
 
-  when(messages("cf.account.details.previous-eori", eoriNumber))
```
Comment (tyagikumarvineet) [inline test/viewmodels/SecuritiesRequestedStatementsViewModelSpec.scala:265]: Why this has been deleted?

File: test/views/components/ErrorSummarySpec.scala [hunk 6]
```diff
@@ -159,13 +155,12 @@ class ErrorSummarySpec extends SpecBase {
...
-        val mockGovSummary: GovukErrorSummary = mock[GovukErrorSummary]
+  trait SetUp {
+    implicit val msgs: Messages = Helpers.stubMessages()
+    val mockGovSummary: GovukErrorSummary = mock[GovukErrorSummary]
 
-        val app = applicationBuilder().overrides(
-            bind[GovukErrorSummary].toInstance(mockGovSummary)
-            ).build()
-    }
+    val app = applicationBuilder().overrides(
```
Comment (tyagikumarvineet) [inline test/views/components/ErrorSummarySpec.scala:162]: Please add an explicit return type as it is a public member

File: app/handlers/ErrorHandler.scala
Comment (tyagikumarvineet) [inline app/handlers/ErrorHandler.scala:1]: Could you please add a test class for these changes? You could see the examples in other repos where we have done the Scala 3 upgrade. If takes a long time for you then leave it

File: app/connectors/CustomsDataStoreConnector.scala [hunk 7]
```diff
@@ -23,32 +23,37 @@ import play.api.http.Status.NOT_FOUND
...
-      case UpstreamErrorResponse(_, NOT_FOUND, _, _) => Left(UnverifiedEmail)
-    }
+    httpClient.get(url"$dataStoreEndpoint")
+      .execute[EmailResponse]
+      .map {
+        case EmailResponse(Some(address), _, None) => Right(Email(address))
+        case EmailResponse(Some(email), _, Some(_)) => Left(UndeliverableEmail(email))
+        case _ => Left(UnverifiedEmail)
+      }.recover {
+        case UpstreamErrorResponse(_, NOT_FOUND, _, _) => Left(UnverifiedEmail)
```
[the remark this answers is not in the document]
Comment (HariHmrc) [reply app/connectors/CustomsDataStoreConnector.scala:46]: corrected the intend and already formatted via intelliJ

File: app/repositories/SessionRepository.scala [hunk 8]
```diff
@@ -53,9 +55,9 @@ class DefaultSessionRepository @Inject()(mongoComponent: PlayMongoComponent, con
 
   override def set(userAnswers: UserAnswers): Future[Boolean] =
     collection.replaceOne(
-      equal("_id", userAnswers.id),
-      userAnswers.copy(lastUpdated = LocalDateTime.now()),
-      ReplaceOptions().upsert(true))
+        equal("_id", userAnswers.id),
```
[the remark this answers is not in the document]
Comment (HariHmrc) [reply app/repositories/SessionRepository.scala:58]: corrected the intend, and formatting already done via intelliJ

File: test/viewmodels/SecuritiesRequestedStatementsViewModelSpec.scala (hunk 5 above)
Comment (HariHmrc) [reply test/viewmodels/SecuritiesRequestedStatementsViewModelSpec.scala:265]: mocking was redundant, without mocking the test case works fine without troubles

**Pull request merged**